x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -116,14 +116,18 @@ export class TaskRunner<
     // Only fetch encrypted attributes here, we'll create a saved objects client
     // scoped with the API key to fetch the remaining data.
     const {
-      attributes: { apiKey },
+      attributes,
     } = await this.context.encryptedSavedObjectsClient.getDecryptedAsInternalUser<RawAlert>(
       'alert',
       alertId,
       { namespace }
     );
 
-    return apiKey;
+    if (!attributes.hasOwnProperty('apiKey')) {
```

Comment:
Sounds like you're testing with no security right now (IRL), so I guess we'll find out if we have this property set for "no security" scenarios!  I'm guessing there's a good chance we never set the field, and so we might not have that property, and so we wouldn't want to throw an error, but return null instead.  Not sure.

Reply:
That was just a quick look in the task_runner - I'll do a full re-review though ...

Reply:
@pmuellr Finally was able to verify that with security turned off, the apiKey field exists and is set to null. I ran my test migration after changing the encryption key and verified that there were no decryption failures during the migration (because there's nothing to decrypt) and so the apiKey field remains null and the rule runs fine after migration.

That being said, I am going to wait for [this PR](https://github.com/elastic/kibana/pull/105333) to be merged and test again after that. It seems like that should fix the unhandled promise rejection and then it won't matter if the api key is null or undefined.